### PR TITLE
Timestamp handling

### PR DESCRIPTION
**Pull request by samirketema**
### Description

Please include a summary of the change and what type of change it is (new feature, bug fix, refactoring, documentation).
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

### Quick checks:

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-snowflake/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination/format/csv.go [hunk 1]
```diff
@@ -21,26 +21,117 @@ import (
...
+	SnowflakeGeography = "GEOGRAPHY"
+	SnowflakeGeometry  = "GEOMETRY"
+
+	// vector data types
+	SnowflakeVector = "VECTOR"
+)
+
+// Map from Avro Types to Snowflake Types
+// TODO: be more precise about numeric types.
+var AvroToSnowflakeType = map[string]string{
```
Comment (lyuboxa) [inline destination/format/csv.go:90]:
```suggestion
var AvroToSnowflakeType = map[avro.Type]string{
```

and then map the simple types. `avro.Boolean: SnowflakeBoolean` potentially.

File: destination/format/csv.go [hunk 2]
```diff
@@ -21,26 +21,117 @@ import (
...
-	snowflakeBoolean   = "BOOLEAN"
-	snowflakeFloat     = "FLOAT"
+	AvroBoolean         = "boolean"
+	AvroInt             = "int"
+	AvroLong            = "long"
+	AvroFloat           = "float"
+	AvroDouble          = "double"
+	AvroBytes           = "bytes"
+	AvroString          = "string"
+	AvroDecimal         = "decimal"
```
Comment (lyuboxa) [inline destination/format/csv.go:119]: there isn't an explicit decimal type, but rather `fixed` with logical subtype decimal

File: destination/format/csv.go [hunk 3]
```diff
@@ -21,26 +21,117 @@ import (
...
+	"strconv"
+	"time"
 
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	"github.com/go-errors/errors"
+	"github.com/hamba/avro"
 	"golang.org/x/exp/maps"
 )
 
+const YYYYMMDD = "2006-01-02"
```
Comment (lyuboxa) [inline destination/format/csv.go:33]: This is the equivalent of `time.DateOnly` const

File: destination/format/csv.go (hunk 2 above)
Comment (samirketema) [reply destination/format/csv.go:119]: yeah, but isn't this the same with `date`?

Comment (lyuboxa) [reply destination/format/csv.go:119]: kind of but not the same, dates/timestamps are represented as long, you will need to look in the logical subtype to determine what is it.  These need to be handle separately like this, won't be a straight forward map resolver.

```
p := field.Type().(*avro.PrimitiveSchema)
switch t := p.Logical().(type) {
case avro.Date:
 ....
case ...

}
```

File: destination/format/csv.go [hunk 4]
```diff
@@ -489,3 +478,24 @@ func isDateOrTimeType(in string) bool {
...
+func mapAvroToSnowflake(field *avro.Field) string {
+	p := field.Type().(*avro.PrimitiveSchema)
+	switch p.Logical().Type() {
+	case avro.Decimal:
+		return SnowflakeFloat
+	case avro.UUID:
+		return SnowflakeVarchar
+	case avro.Date:
+		return SnowflakeDate
+	case avro.TimeMillis:
```
Comment (lyuboxa) [inline destination/format/csv.go:491]: Pg also sends `LocalTimestampMicros` which is I think SnowflakeTimestampNTZ (I think this is the local time variety in snowflake?)

Comment (samirketema) [reply destination/format/csv.go:491]: that is covered by `avro.TimeMicros` right? unless you mean we need to pass down the timezone for timezone-specific scenarios

Comment (lyuboxa) [reply destination/format/csv.go:491]: I think it maps to different types. The local timestamp in avro does not carry a TZ component.  Timestamp with TZ and timestamp without TZ

File: destination/format/csv.go [hunk 5]
```diff
@@ -489,3 +478,24 @@ func isDateOrTimeType(in string) bool {
 		return false
 	}
 }
+
+func mapAvroToSnowflake(field *avro.Field) string {
+	p := field.Type().(*avro.PrimitiveSchema)
+	switch p.Logical().Type() {
```
Comment (lyuboxa) [inline destination/format/csv.go:484]: Can Logical() return nil?

Comment (samirketema) [reply destination/format/csv.go:485]: yeah, I just got a nil pointer here 🤣 gonna fix this, thanks

**Comment by raulb** [comment]
@samirketema I opened this other PR to fix what's missing on this one https://github.com/conduitio-labs/conduit-connector-snowflake/pull/119

**Review by raulb — APPROVED** [review]

**Pull request merged**